src/main.cpp

```diff
@@ -62,6 +75,37 @@ class Demo {
 			return -1;
 		}
 
+		if (debugMode == 1) {
+			initDebugTextures();
+		}
+
+		audioSpec.freq = 44100;
+		audioSpec.format = AUDIO_F32SYS;
+		audioSpec.channels = 2;
+		audioSpec.samples = 2048;
+		audioSpec.callback = fillAudioBuffer;
+		audioSpec.userdata = &dmg;
+		
+		// TODO: Investigate audio device flags
+		audioDevice = SDL_OpenAudioDevice(nullptr, 0, &audioSpec, nullptr, 0);
+		
+		if (audioDevice == NULL) {
+			std::cout << "Audio device error. SDL_Error: " << SDL_GetError() << std::endl;
+			close();
+			return -1;
+		}
+
+		SDL_PauseAudioDevice(audioDevice, 0);
+
+		// Start dmg
+		dmg.init();
+
+		return 0;
+	}
+
+	int initDebugTextures() {
```

Comment:
I feel like its better to pull debug related stuff into their own functions where it's convenient. Not super sure the best way to separate out debug related things though, could maybe use another pass at some point.